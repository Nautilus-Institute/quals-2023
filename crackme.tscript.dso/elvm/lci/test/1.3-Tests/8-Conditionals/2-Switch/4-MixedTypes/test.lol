HAI 1.3
	1
	WTF?
		OMG WIN
			VISIBLE "1a"
		OMG 1.0
			VISIBLE "1b"
		OMG "1"
			VISIBLE "1c"
		OMG 1
			VISIBLE "1d"
	OIC
	0.0
	WTF?
		OMG FAIL
			VISIBLE "2a"
		OMG 0
			VISIBLE "2b"
		OMG "0"
			VISIBLE "2c"
		OMG 0.0
			VISIBLE "2d"
	OIC
	"0"
	WTF?
		OMG FAIL
			VISIBLE "3a"
			GTFO
		OMG 0
			VISIBLE "3b"
			GTFO
		OMG 0.0
			VISIBLE "3c"
			GTFO
		OMGWTF
			VISIBLE "3d"
	OIC
KTHXBYE
