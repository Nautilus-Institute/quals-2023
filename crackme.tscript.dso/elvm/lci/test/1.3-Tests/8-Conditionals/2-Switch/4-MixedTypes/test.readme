This test checks that mixed literal types may appear within the case block.
Note that the spec is unclear as to whether other types are implicitly cast to
the type of the implicit variable.  This test assumes the same semantics as the
built in comparison operator, namely, a guard matches the implicit variable if
both its type and value match.
