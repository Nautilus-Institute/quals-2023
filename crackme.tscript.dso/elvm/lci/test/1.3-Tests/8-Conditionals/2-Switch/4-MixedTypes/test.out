1d
2d
3d
